resync CSS selector parsing tests

keithamus opened the pull request:
#### e694f63e91aad88d879b8730a588972b8f4a0ec2
<pre>
resync CSS selector parsing tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=265785">https://bugs.webkit.org/show_bug.cgi?id=265785</a>

Reviewed by Tim Nguyen.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/5ddf095c380db657c000017e46dfebc45e93f682">https://github.com/web-platform-tests/wpt/commit/5ddf095c380db657c000017e46dfebc45e93f682</a>

* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-forgiving-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-forgiving-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-part-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-part.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-slotted-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-slotted.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/271484@main">https://commits.webkit.org/271484@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dea0fdacab1e128b6994459862b3b6b8e8f52bea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28529 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7173 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29916 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31057 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25983 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9282 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4542 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/26183 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28799 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5930 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24544 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5181 "Found 1 new API test failure: /WPE/TestWebKitWebView:/webkit/WebKitWebView/notification (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5307 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25544 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31757 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26130 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25986 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31575 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5269 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3440 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29347 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6871 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/25338 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6834 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5727 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5789 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
You'll need to rebaseline the tests too.

keithamus committed:
resync CSS selector parsing tests
https://bugs.webkit.org/show_bug.cgi?id=265785

Reviewed by Tim Nguyen.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/5ddf095c380db657c000017e46dfebc45e93f682

* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-forgiving-selector-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has-forgiving-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-has.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-part-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-part.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-slotted-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-slotted.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state-expected.txt: Rebaselined.
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/parse-state.html:
* LayoutTests/imported/w3c/web-platform-tests/css/selectors/parsing/w3c-import.log:

Canonical link: https://commits.webkit.org/271484@main
